### PR TITLE
from-handlers package

### PR DESCRIPTION
**Pull request by rpominov**
Part of #84  

## Description

Added a `@webstudio-is/form-handlers` package that we're going to use to send submitted forms as emails. 

There're two handlers currently:

 - mailchanels (https://blog.cloudflare.com/sending-email-from-workers-with-mailchannels/)
 - n8n

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
- [ ] hi @istarkov, I need you to do
  - conceptual review (architecture, feature-correctness)

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/mailer/src/send-email.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+type Personalization = {
+  to: [EmailAddress, ...EmailAddress[]];
+  from?: EmailAddress;
+  dkim_domain?: string;
+  dkim_private_key?: string;
+  dkim_selector?: string;
+  reply_to?: EmailAddress;
+  cc?: EmailAddress[];
+  bcc?: EmailAddress[];
```
Comment (TrySound) [inline packages/mailer/src/send-email.ts:18]: I have flashbacks 🤯

**Comment by rpominov** [comment]
Ready for review again

File: packages/form-handlers-mailchannels-test/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@webstudio-is/form-handlers-mailchannels-test",
+  "private": true,
+  "version": "0.0.0",
+  "scripts": {
+    "deploy": "wrangler publish src/index.ts",
+    "dev": "wrangler dev src/index.ts"
```
Comment (istarkov) [inline packages/form-handlers-mailchannels-test/package.json:7]: Am I right, if I execute pnpm run dev at tge root this will be executed too?

Comment (rpominov) [reply packages/form-handlers-mailchannels-test/package.json:7]: Yes. Fixing

File: packages/form-handlers-mailchannels-test/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export default {
+  async fetch(request: Request) {
+    const url = new URL(request.url);
+
+    const to = url.searchParams.get("to");
+
+    if (to == null) {
+      return new Response(
+        `Set "to" query parameter to your email. Example: https://...?to=bob@example.com`
```
Comment (istarkov) [inline packages/form-handlers-mailchannels-test/src/index.ts:11]: to to mb like:
Set the "to" query parameter with your email, like this: https://.../?to=bob@example.com

File: packages/form-handlers-mailchannels-test/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      );
+    }
+
+    const formData = new FormData();
+    formData.append("name", "Bob");
+    formData.append("time", new Date().toLocaleString("en-US"));
+
+    const result = await mailchannelsHandler({
+      senderEmail: "test@webstudio.is",
```
Comment (istarkov) [inline packages/form-handlers-mailchannels-test/src/index.ts:20]: mb `@example.com` ?

File: packages/form-handlers/src/mailchanels.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  subject?: string;
+  headers?: Record<string, string>;
+};
+
+type ContentItem = {
+  type: "text/plain" | "text/html";
+  value: string;
+};
+
+type SendEmailPayload = {
```
Comment (istarkov) [inline packages/form-handlers/src/mailchanels.ts:32]: Comment plz from where does this types same from. I mean API href, doc href etc

File: packages/form-handlers-mailchannels-test/src/index.ts (hunk 3 above)
Comment (rpominov) [reply packages/form-handlers-mailchannels-test/src/index.ts:11]: This is just a package for developers to test, doesn't matter either way

File: packages/form-handlers/src/mailchanels.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+const sendEmail = async (
+  payload: SendEmailPayload
+): Promise<SendEmailResult> => {
+  const response = await fetch("https://api.mailchannels.net/tx/v1/send", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(payload),
+  });
+
+  if (response.status === 202) {
```
Comment (istarkov) [inline packages/form-handlers/src/mailchanels.ts:52]: Why 202, why 200 is not suitable, why any 2xx is not suitable

File: packages/form-handlers-mailchannels-test/src/index.ts (hunk 4 above)
Comment (rpominov) [reply packages/form-handlers-mailchannels-test/src/index.ts:20]: Mailchanels doesn't allow to send from `example.com` 🤷‍♂️

File: packages/form-handlers/src/mailchanels.ts (hunk 5 above)
Comment (rpominov) [reply packages/form-handlers/src/mailchanels.ts:32]: There's comment at top of the file

File: packages/form-handlers/src/mailchanels.ts (hunk 6 above)
Comment (rpominov) [reply packages/form-handlers/src/mailchanels.ts:52]: This is what mailchanels respond with. Copied from https://github.com/cloudflare/pages-plugins/blob/c9bdd1e8a10ad276958c399111a1e5293574af4f/packages/mailchannels/api/index.ts#L53

File: packages/form-handlers/src/mailchanels.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    return { success: true };
+  }
+
+  try {
+    const { errors } = await response.json();
+    return { success: false, errors };
+  } catch {
+    return { success: false, errors: [response.statusText] };
```
Comment (istarkov) [inline packages/form-handlers/src/mailchanels.ts:60]: never seen statusText in the wild. 
Usually to get something what can be used.

const text = await response.text();
const res = zodError(text);
if (res.success) { 
...
} 
{sucsess: false, "errors": [text.slice(..., N)]

Comment (rpominov) [reply packages/form-handlers/src/mailchanels.ts:60]: Also took from https://github.com/cloudflare/pages-plugins/blob/c9bdd1e8a10ad276958c399111a1e5293574af4f/packages/mailchannels/api/index.ts#L59

**Review by istarkov — APPROVED** [review]

**Pull request merged**